### PR TITLE
[WIP] Runner smoketests

### PR DESCRIPTION
**Pull request by jctanner**
##### SUMMARY
Implements basic smoke tests for ansible-runner so that we can ensure compatibility with every PR to ansible/ansible.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
tests

##### ANSIBLE VERSION

```
2.7
```

##### ADDITIONAL INFORMATION
Requires https://github.com/ansible/ansible-runner-test-container

### CODE REVIEW

File: test/integration/targets/ansible_runner_smoketests/files/adhoc_example1.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+data = {
+    'rc': r.rc,
+    'status': r.status,
+    'events': [x['event'] for x in r.events],
+    'stats': r.stats
+}
+
+# insert this header for the flask controller
+print('#STARTJSON')
+print(json.dumps(data))
```
Comment (webknjaz) [inline test/integration/targets/ansible_runner_smoketests/files/adhoc_example1.py]: `json.dump(data, sys.stdout)`

File: test/integration/targets/ansible_runner_smoketests/files/playbook_example1.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    os.makedirs(invdir)
+with open(os.path.join(invdir, 'hosts'), 'w') as f:
+    f.write('localhost\n')
+pbfile = os.path.join(output_dir, 'test.yml')
+with open(pbfile, 'w') as f:
+    f.write(PLAYBOOK)
+
+r = ansible_runner.run(private_data_dir=output_dir, playbook='test.yml')
+
+data = {
```
Comment (webknjaz) [inline test/integration/targets/ansible_runner_smoketests/files/playbook_example1.py:31]: This duplicates previous module converter, it makes sense to put it in a shared utils module

File: test/integration/targets/ansible_runner_smoketests/tasks/adhoc_example1.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+- name: send the playbook script
+  shell: "{{ 'curl -X POST -F \"file=@' + role_path + '/files/' + 'adhoc_example1.py\" ' + runner_container_url + '/upload_file' }}"
```
Comment (webknjaz) [inline test/integration/targets/ansible_runner_smoketests/tasks/adhoc_example1.yml]: To reduce need escaping you can use multiline block:
```yaml
  shell: >
    {{ 'curl -X POST -F "file=@' + role_path + '/files/' + 'adhoc_example1.py" ' + runner_container_url + '/upload_file' }}
```

File: test/integration/targets/ansible_runner_smoketests/tasks/setup.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+- name: get the container host and port
+  set_fact:
+    runner_container_host: "{{ lookup('env', 'ANSIBLE_RUNNER_HOST') }}"
+    runner_container_port: "{{ lookup('env', 'ANSIBLE_RUNNER_PORT') }}"
+    runner_container_url: "{{ 'http://' + lookup('env', 'ANSIBLE_RUNNER_HOST') + ':' + lookup('env', 'ANSIBLE_RUNNER_PORT') }}"
+- name: check for ansible install path on container
+  uri:
+    url: "{{ runner_container_url + '/' + 'run_command' }}"
```
Comment (webknjaz) [inline test/integration/targets/ansible_runner_smoketests/tasks/setup.yml]: `'/' + 'run_command'` -> `'/run_command'`

File: test/integration/targets/ansible_runner_smoketests/files/playbook_example1.py (hunk 2 above)
Comment (jctanner) [reply test/integration/targets/ansible_runner_smoketests/files/playbook_example1.py:31]: These are standalone scripts that are executed inside the runner container, so I was avoiding utils based purely on the fact that i didn't want to ship more than one file over to the container for each test. I could theoretically zip/tar up a couple things on the control node and send that over to the container instead, but that's more work that I wanted to do upfront.

**Comment by jctanner** [comment]
Closing in favor of https://github.com/ansible/ansible/pull/44803

**Pull request closed without merging**